superset/connectors/sqla/models.py

```diff
@@ -303,13 +307,15 @@ def get_timestamp_expression(
 
         pdf = self.python_date_format
         is_epoch = pdf in ("epoch_s", "epoch_ms")
+        column_spec = self.db_engine_spec.get_column_spec(self.type)
+        type_ = column_spec.sqla_type if column_spec else DateTime
         if not self.expression and not time_grain and not is_epoch:
-            sqla_col = column(self.column_name, type_=DateTime)
+            sqla_col = column(self.column_name, type_=type_)
```

Comment:
This was incorrectly hardcoded to `DateTime` (the correct type is received from the column spec)

Reply:
is there a test for this change?

Reply:
Good idea; I added tests for this (now the type is the "TemporalWrapperType" which is asserted for in the `test_get_timestamp_expression` test)